src/PlexRequests.Api/ApiService.cs

```diff
@@ -1,27 +1,40 @@
 ﻿using System.Collections.Generic;
 using System.IO;
+using System.Net;
 using System.Net.Http;
 using System.Threading.Tasks;
 using System.Xml.Serialization;
+using Microsoft.Extensions.Logging;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Serialization;
+using PlexRequests.Helpers;
 
 namespace PlexRequests.Api
 {
     public class ApiService : IApiService
     {
         private readonly IPlexRequestsHttpClient _httpClient;
+        private readonly ILogger<ApiService> _logger;
 
-        public ApiService(IPlexRequestsHttpClient httpClient)
+        public ApiService(
+            IPlexRequestsHttpClient httpClient,
+            ILogger<ApiService> logger
+            )
         {
             _httpClient = httpClient;
+            _logger = logger;
         }
 
         public async Task InvokeApiAsync(ApiRequest request)
         {
             using (var httpRequestMessage = CreateHttpRequestMessage(request))
             {
-                await _httpClient.SendAsync(httpRequestMessage);
+                var httpResponse =await _httpClient.SendAsync(httpRequestMessage);
+
+                if (!httpResponse.IsSuccessStatusCode)
+                {
+                    await LogApiUnsuccessful(request, httpResponse);
```

Comment:
Do you really want to log all unsuccessful api calls? May be filter to 500 errors

Reply:
Yeh. These are downstream API's. So Plex might return a useful error on a 400+ range and i'll log that.